pointsource_imaging/MIRI_Aperture_Photometry_solution.ipynb

```diff
@@ -161,6 +161,18 @@
     "### 2.2<font color='white'>-</font>Plot configuration<a class=\"anchor\" id=\"plot\"></a> ###"
```

Comment:
<p>Since it takes a little bit to download these data files, it would be good to check if the folders already exist before downloading them. Here's how I would do it:</p><p> </p><p>1. Move <code>folder_path</code>'s assignment to this cell.</p><p>2. Place all of the unzipped files into a deeper directory instead of the notebook's current directory. In essence, change <code>folder_path</code> in the cell after the download from <code>./</code>to <code>./FOLDERNAME/</code>.</p><p>3. Check if the path to <code>FOLDERNAME</code> already exists.</p><p>4. If not, check if the zip files already exist.</p><p>5. If not, _then_ do the download from Box.</p><p> </p><p>It could look something like this:</p><pre class="ql-syntax" spellcheck="false">folder_path = './FOLDERNAME/'

if os.path.isdir(folder_path):
&nbsp;&nbsp;&nbsp; pass
elif ( os.path.isfile('./stage2-selected.zip')
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; and os.path.isfile('./stage3-selected.zip') ):
&nbsp;&nbsp;&nbsp; # extract both zip files if they're present
&nbsp;&nbsp;&nbsp; with zipfile...
else:
&nbsp;&nbsp;&nbsp; # do both downloads
&nbsp;&nbsp;&nbsp; boxlink = ...
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jwebbinar_prep/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Reply:
I changed this cell as you suggested. Now it should check if the folder exists first, then if the zip files are present, and only as last option download the data. I added the zip-file check in the first if just in case someone set FOLDERNAME equal to ''